src/buffer/out/OutputCellIterator.cpp

```diff
@@ -206,6 +206,40 @@ OutputCellIterator& OutputCellIterator::operator++()
     // Keep track of total distance moved (cells filled)
     _distance++;
 
+    if (_mode == Mode::Loose && _currentView.DbcsAttr().IsSingle())
+    {
+        size_t pos = _pos;
+        pos += _currentView.Chars().size();
+        const std::wstring_view run = std::get<std::wstring_view>(_run);
+        if (pos < run.length())
+        {
+            const wchar_t wch = run.at(pos);
+            if (0x20 <= wch && wch <= 0x7e)
+            {
+                _currentView.UpdateText(run.substr(pos, 1));

```

Comment:
Again, inspired by #10621, use `UpdateText` instead of calling expensive constructors.